clrenv/path.py

```diff
@@ -1,32 +1,74 @@
-import os.path
+"""
+Manages environment yaml file paths.
+
+clrenv expects a base environment file (environment.yaml) and zero or more
+overlay files.
+
+Without configuration (via env vars) looks for an environment.yaml in the cwd
+or one of its parents. This can be set explicitly by setting CLRENV_PATH.
+
+Overlay files (values will mask those set in the base file) can be specified
+with the CLRENV_OVERLAY_PATH env var. Multiple overlays can be specified with a
+colon (:) delimiter. Overlay files that do not exist will be ignored.
+"""
+from pathlib import Path
 from os import environ
+import logging
 
+logger = logging.getLogger(__name__)
 
-CLRENV_PATH = environ.get("CLRENV_PATH")
-CLRENV_OVERLAY_PATH = environ.get("CLRENV_OVERLAY_PATH")
+DEBUG_MODE = environ.get("CLRENV_DEBUG", "").lower() in ("true", "1")
 
 
-def find_environment_path(name="environment.yaml"):
-    if CLRENV_PATH:
-        return CLRENV_PATH
-    path = _recursively_find_file_path(name)
+def _resolve_path(path):
+    """Resolves a user provided path. Returns None for empty string.
+
+    Does not confirm the file exists.
+    """
     if not path:
-        raise Exception("%s could not be located." % name)
-    return path
+        return
+    return Path(path).expanduser().absolute()
+
+
+def _resolve_paths(paths):
+    """Resolves a user provided, colon seperated, list of paths.
+
+    Will drop values that do not exist."""
+    result = []
+    if not paths:
+        return result
```

Comment:
nit: Might be a way to only need one `return result` statement, i.e. gracefully manage `paths=''` or `paths=None` in L41?